packages/calypso-e2e/src/lib/pages/reader-page.ts

```diff
@@ -70,22 +64,33 @@ export class ReaderPage {
 			throw new Error( 'Unable to select and visit post - specify one of index or text.' );
 		}
 
-		await Promise.all( [ this.page.waitForNavigation(), this.page.click( selector ) ] );
+		await Promise.all( [
+			this.page.waitForSelector( selectors.placeholder, { state: 'hidden' } ),
+			this.page.waitForNavigation( { waitUntil: 'networkidle' } ),
+			this.page.click( selector ),
```

Comment:
I think I'm a little bit confused by this order...

Which placeholder is this referring to? Is this on the reader feed page with lots of posts? Or is this for a specific post after you click on a specific article?

If it's for the former, I think we want to full wait for that before clicking on the selector.

Reply:
Right, I understand where the confusion is coming from.

The intention here is for the following to occur:
- click on the selector, which points to a post in the main Reader stream.
- wait for the subsequent navigation to fully complete **AND** wait for all placeholder elements to disappear.

The placeholder elements show up in this section when a particular post is visited:
![image](https://user-images.githubusercontent.com/6549265/140416712-10f70370-3b51-48d1-9dc3-1933925fa9c5.png)

Now, you are right that there are multiple levels of placeholders that are observed in the Reader stream feature:
- when loading the reader stream.
- when visiting a post.

Given this I will rename the `placeholder` selector to something a bit more descriptive.

Reply:
I've made some changes in 6c2f99595584aee1ed1ee63c318270f3c21f6833. Let me know what you think!